### PR TITLE
MPDX-7030-Add-Autocomplete-To-ContactTags

### PR DESCRIPTION
**Pull request by TheNoodleMoose**
Had to do this autocomplete component a little bit different than how the others are used throughout the app because of the design for contact tags. But happy with how it turned out. Feel free to try it out and let me know what you think! If you see anything weird in the code, please ask questions 🙂

### CODE REVIEW

**Review by LizzyRandall — COMMENTED** [review]
Looks good to me

**Review by gyasistory — APPROVED** [review]

File: src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx [hunk 1]
```diff
@@ -95,24 +99,25 @@ export const ContactTags: React.FC<ContactTagsProps> = ({
   };
 
   const onSubmit = async (
-    { tag }: { tag: string },
-    { resetForm }: FormikHelpers<{ tag: string }>,
+    { tagList }: { tagList: string[] & never[] },
```
Comment (OzzieOrca) [inline src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:102]: What's the `& never[]` about?

**Review by OzzieOrca — APPROVED** [review]

File: src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx (hunk 1 above)
Comment (TheNoodleMoose) [reply src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:102]: So every time the user adds new tags, I want to reset the initial Formik value of ```tagList``` back to an empty array:
```
     <Formik
        initialValues={{ tagList: [] }}
        validationSchema={tagSchema}
        onSubmit={onSubmit}
      >
```
So it's inheriting the type of ```never[]``` from that initial value, plus the ```tagList: string[]``` I set.

Comment (OzzieOrca) [reply src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:102]: Were the types complaining? `string[]` can also be an empty array. But this if fine if it's needed to work :) I'm not familiar with Formik's typings.

Comment (TheNoodleMoose) [reply src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx:102]: It complained in the onSubmit function. That's what I thought too, but TS just kept yelling 🤷‍♂️

**Pull request merged**